### PR TITLE
fix: emit SelfServiceMethodUsed in SettingsSucceeded event

### PR DESCRIPTION
**Pull request by hperl**
This PR fixes the SelfServiceMethodUsed attribute in the SettingsSucceeded event, which previously was always empty.

## Checklist

- [ ] I have read the [contributing guidelines](../blob/master/CONTRIBUTING.md).
- [ ] I have referenced an issue containing the design document if my change
      introduces a new feature.
- [ ] I am following the
      [contributing code guidelines](../blob/master/CONTRIBUTING.md#contributing-code).
- [ ] I have read the [security policy](../security/policy).
- [ ] I confirm that this pull request does not address a security
      vulnerability. If this pull request addresses a security vulnerability, I
      confirm that I got the approval (please contact
      [security@ory.sh](mailto:security@ory.sh)) from the maintainers to push
      the changes.
- [ ] I have added tests that prove my fix is effective or that my feature
      works.
- [ ] I have added or changed [the documentation](https://github.com/ory/docs).

### CODE REVIEW

File: selfservice/flow/settings/hook.go [hunk 1]
```diff
@@ -282,7 +282,8 @@ func (e *HookExecutor) PostSettingsHook(w http.ResponseWriter, r *http.Request,
 		WithField("flow_method", settingsType).
 		Debug("Completed all PostSettingsPrePersistHooks and PostSettingsPostPersistHooks.")
 
-	trace.SpanFromContext(r.Context()).AddEvent(events.NewSettingsSucceeded(r.Context(), i.ID, string(ctxUpdate.Flow.Type), ctxUpdate.Flow.Active.String()))
+	trace.SpanFromContext(r.Context()).AddEvent(events.NewSettingsSucceeded(
+		r.Context(), i.ID, string(ctxUpdate.Flow.Type), ctxUpdate.Session.AMR))
```
Comment (jonas-jonas) [inline selfservice/flow/settings/hook.go:286]: Isn't `ctxUpdate.Session.AMR` the method used to authenticate the session?

I would expect that field to contain `"profile"`, `"password"`, etc. depending on which group the user updated in the settings.

Comment (hperl) [reply selfservice/flow/settings/hook.go:286]: Yes, it is the method used in the session. Should it be the group that got updated? Maybe that makes more sense, but it is not clear.

Comment (jonas-jonas) [reply selfservice/flow/settings/hook.go:286]: Yes, I think it should be.

**Review by jonas-jonas — APPROVED** [review]
Brilliant!

**Review by aeneasr — APPROVED** [review]

**Pull request merged**